fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -2579,6 +2582,12 @@ ACTOR Future<Void> reply(CommitBatchContext* self) {
 		                                     self->prevVersion,
 		                                     writtenTags),
 		    TaskPriority::ProxyMasterVersionReply));
+		if (SERVER_KNOBS->ENABLE_VERSION_VECTOR_TLOG_UNICAST) {
+			ASSERT(self->loggingComplete.isReady());
```

Comment:
It seems possible this ASSERT would fail if there was a race, where `when(wait(pProxyCommitData->committedVersion.whenAtLeast(self->commitVersion + 1))) {}` was true before `self->loggingComplete` was set. I think you need a wait here, which would be a null op in most cases.

Reply:
I thought it was safe to do the assert here because of this check https://github.com/apple/foundationdb/blob/5497488e411735beb1e52f3ee559cd17fa2258f8/fdbserver/CommitProxyServer.actor.cpp#L2569, which wouldn't be true if https://github.com/apple/foundationdb/blob/5497488e411735beb1e52f3ee559cd17fa2258f8/fdbserver/CommitProxyServer.actor.cpp#L2507 was true.

Reply:
Setting "pProxyCommitData->minKnownCommittedVersion" here might be correct, but could be redundant though. Probably we need to move this code to be in this condition https://github.com/apple/foundationdb/blob/5497488e411735beb1e52f3ee559cd17fa2258f8/fdbserver/CommitProxyServer.actor.cpp#L2589.

Reply:
Done.